compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -284,6 +285,25 @@ impl<'tcx> SizeSkeleton<'tcx> {
                     ),
                 }
             }
+            ty::Array(ty, len) if len.ty() == tcx.types.usize => {
+                match SizeSkeleton::compute(ty, tcx, param_env)? {
+                    SizeSkeleton::Known(s) => {
+                        if let Some(c) = len.try_eval_usize(tcx, param_env) {
+                            Ok(SizeSkeleton::Known(c * s))
```

Comment:
This should be a `bug!` (`tcx.layout_of` call above should've succeeded if the length is constant, see the existing `bug!` just a few lines above).

Reply:
I think this can still be hit (I hit it while testing overflow), when the `layout_of` overflows for a nested array `[[u32; 99999999]; 99999999]`, but we do not overflow for just the outer array. Eventually we will get an overflow, but we don't want to throw a bug here